api/envoy/config/filter/network/sni_dynamic_forward_proxy/v2alpha/sni_dynamic_forward_proxy.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package envoy.config.filter.network.sni_dynamic_forward_proxy.v2alpha;
+
+import "envoy/config/common/dynamic_forward_proxy/v2alpha/dns_cache.proto";
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.network.sni_dynamic_forward_proxy.v2alpha";
+option java_outer_classname = "SniDynamicForwardProxyProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.filters.network.sni_dynamic_forward_proxy.v3alpha";
+
+// [#protodoc-title: SNI dynamic forward proxy]
+
+// Configuration for the SNI-based dynamic forward proxy filter. See the
+// :ref:`architecture overview <arch_overview_http_dynamic_forward_proxy>` for
+// more information. Note this filter must be configured along with
+// :ref:`TLS inspector listener filter <config_listener_filters_tls_inspector>` to work.
+// [#extension: envoy.filters.network.sni_dynamic_forward_proxy]
+message FilterConfig {
+  // The DNS cache configuration that the filter will attach to. Note this
+  // configuration must match that of associated :ref:`dynamic forward proxy
+  // cluster configuration
+  // <envoy_api_field_config.cluster.dynamic_forward_proxy.v2alpha.ClusterConfig.dns_cache_config>`.
+  common.dynamic_forward_proxy.v2alpha.DnsCacheConfig dns_cache_config = 1
```

Comment:
Then where it loads config from? cluster? Filter doesn't know where the cluster when it is configured.

Reply:
I see. This is another annoying instance in the API where we expect duplicated configuration because Envoy internally doesn't have some way to reconcile indirections. I think this is worth thinking about in the context of UDPA,